airbag: T2088: do not call Config.in_session

thomas-mangin opened the pull request:
It was found that calling Config() was extremely slow and
therefore was affecting performance. It was called by airbag at
each import. The first band-aid is to only call the function if
the router if fully loaded and detect boot using vyos-config-status

For clarity, it would only run once even if imported multiple time as the module is only initialised once and then referenced when imported.

jestabro commented:
The issue stills remains with this patch; timing is unaffected (adding 100 dummy interfaces):

with airbag (same as before):
vyos@vyos# time commit

real    0m37.030s
user    0m16.931s
sys     0m7.978s

without:
vyos@vyos# time commit

real    0m26.742s
user    0m13.513s
sys     0m5.136s

Also, although it is true that the module is referenced after the first import, one has this import in all _processes_ which have it in the path, through its inclusion via vrrp.py or util.py.

thomas-mangin committed:
airbag: T2088: do not call Config.in_session

It was found that calling Config() was extremely slow and
therefore was affecting peformance. It was called by airbag at
each import. The first band-aid is to only call the function if
the router if fully loaded and detect boot using vyos-config-status

thomas-mangin commented:
patch changed to have False set instead of True .. disabling at boot (does not change the code path for when Config() is called) but it should not be called until the end of the boot.

Looking at the system trace we used / have to see two calls to cli-shell-api at boot when the `__init__` of Config is called. That init should not be called by airbag at boot by the patch now.

However, all the other commit command will still Config and will have the slow down unless https://github.com/vyos/vyos-1x/pull/424 was applied, then it is still called but not as slow.

c-po commented:
Is this different from Config.in_session?

https://github.com/vyos/vyos-1x/blob/current/python/vyos/config.py#L223

jestabro commented:
Not different; that is the call that is causing the slow down.

thomas-mangin commented:
Hum, ... I need to look at how to generate the systemd trace with this patch

c-po commented:
It is described here: https://docs.vyos.io/en/latest/contributing/development.html#boot-timing

thomas-mangin commented:
This is withdrawn and #439 is proposed instead.
